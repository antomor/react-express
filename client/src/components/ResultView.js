import React, {Component} from 'react';
import PropTypes from "prop-types";
import DuplicateList from './DuplicateList';
import Error from './Error';

class ResultView extends Component{
  render(){
    const msg = "Something wrong with count"
    const view = !this.props.items ? 
      <Error msg={msg}/>:
      <DuplicateList items={this.props.items} />
    return (
      <div>
        {view}
      </div>
    )
  }
}

ResultView.propTypes = {
  items: PropTypes.arrayOf(PropTypes.shape({
    key: PropTypes.string.isRequired,
    value: PropTypes.number.isRequired,
  }))
}

export default ResultView