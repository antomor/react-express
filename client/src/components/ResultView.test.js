import React from 'react';
import ResultView from './ResultView';
import renderer from 'react-test-renderer';

describe('ResultView', () => {
  it('renders the items properly', () => {
    const items = [
      {
        key: 'aa',
        value: 1
      },
      {
        key: 'bb',
        value: 10
      }
    ]
    const component = renderer.create(
      <ResultView items={items}/>,
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders the Error', () => {
    const component = renderer.create(
      <ResultView />,
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

}); 
