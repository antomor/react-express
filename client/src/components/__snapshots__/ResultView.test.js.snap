// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ResultView renders the Error 1`] = `
<div>
  <div>
    An error occurred:
    <div>
      Something wrong with count
    </div>
  </div>
</div>
`;

exports[`ResultView renders the items properly 1`] = `
<div>
  <div
    className="DuplicateList-root-1"
  >
    <ul
      className="MuiGridList-root-4 DuplicateList-gridList-2"
      style={
        Object {
          "margin": -2,
        }
      }
    >
      <li
        className="MuiGridListTile-root-8"
      >
        <div
          className="MuiGridListTile-tile-9"
        >
          <div
            className="MuiPaper-root-12 MuiPaper-shadow2-16 MuiPaper-rounded-13 DuplicateItem-root-5"
          >
            <div
              className="MuiPaper-root-12 MuiPaper-shadow2-16 MuiPaper-rounded-13 DuplicateItem-itemKey-6"
            >
              aa
            </div>
            <div
              className="MuiPaper-root-12 MuiPaper-shadow2-16 MuiPaper-rounded-13 DuplicateItem-itemValue-7"
            >
              1
            </div>
          </div>
        </div>
      </li>
      <li
        className="MuiGridListTile-root-8"
      >
        <div
          className="MuiGridListTile-tile-9"
        >
          <div
            className="MuiPaper-root-12 MuiPaper-shadow2-16 MuiPaper-rounded-13 DuplicateItem-root-5"
          >
            <div
              className="MuiPaper-root-12 MuiPaper-shadow2-16 MuiPaper-rounded-13 DuplicateItem-itemKey-6"
            >
              bb
            </div>
            <div
              className="MuiPaper-root-12 MuiPaper-shadow2-16 MuiPaper-rounded-13 DuplicateItem-itemValue-7"
            >
              10
            </div>
          </div>
        </div>
      </li>
    </ul>
  </div>
</div>
`;
